apps/cowswap-frontend/src/modules/usdAmount/state/usdTokenPricesAtom.ts

```diff
@@ -23,18 +23,27 @@ export const usdTokenPricesAtom = atom((get) => {
 
     acc[tokenAddress] = {
       ...usdPrice,
-      price: usdPrice.price === null ? null : calculatePrice(usdPrice.currency, usdPrice.price),
+      price: calculatePrice(usdPrice.currency, usdPrice.price),
     }
 
     return acc
   }, {})
 })
 
-function calculatePrice(currency: Token, price: number): Price<Token, Token> {
+function calculatePrice(currency: Token, price: string | null): Price<Token, Token> | null {
+  if (!price) {
+    return null
+  }
+
   const usdcToken = USDC[currency.chainId as SupportedChainId]
 
+  if (price.includes('e') || new RegExp('\\.' + '0'.repeat(usdcToken.decimals)).test(price)) {
+    // Less than 1 atom of USDC, cannot create a Price instance
```

Comment:
mm, isn't this too fragile too? converting numbers into strings and expecting them to be in scientific notation...

Reply:
No, it doesn't expect them to be in scientific notation.
It's just in case they are, which might be I think if the value is smaller than 10 decimals.
The main check is for `^0.000000x...` number, which will be less than 1 USDC atom.

Reply:
way better